packages/runtime-dom/src/modules/attrs.ts

```diff
@@ -59,9 +59,9 @@ export function compatCoerceAttr(
 ): boolean {
   if (isEnumeratedAttr(key)) {
     const v2CoercedValue =
-      value === null
+      value === null || value === false || value === 'false'
         ? 'false'
-        : typeof value !== 'boolean' && value !== undefined
+        : value !== undefined
```

Comment:
I think the logic is correct, but it might be easier to understand if the `undefined` check were handled first?